services/management/dbaas/version_service_client.go

```diff
@@ -196,3 +246,72 @@ func (c *VersionServiceClient) GetLatestOperatorVersion(ctx context.Context, pmm
 	latestPXCOperator, err := getLatest(pmmVersionDeps.Matrix.PXCOperator)
 	return latestPXCOperator, latestPSMDBOperator, err
 }
+
+// GetNextOperatorVersion returns operator version that is direct successor of currently installed one.
+// It returns nil if update is not available or error occurred. It does not take PMM version into consideration.
+// We need to upgrade to current + 1 version for upgrade to be successful. So even if dbaas-controller does not support the
+// operator, we need to upgrade to it on our way to supported one.
+func (c *VersionServiceClient) GetNextOperatorVersion(ctx context.Context, operatorType, installedVersion string) (nextOperatorVersion *goversion.Version, err error) {
+	if installedVersion == "" {
+		return
+	}
+	// Get all operator versions
+	params := componentsParams{
+		product: operatorType,
+	}
+	matrix, err := c.Matrix(ctx, params)
+	if err != nil {
+		return
+	}
+	if len(matrix.Versions) == 0 {
+		return
+	}
+
+	// Find next versions if installed.
+	if installedVersion != "" {
+		nextOperatorVersion, err = getNext(matrix.Versions, installedVersion)
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+func getNext(versions []struct {
+	Product        string `json:"product"`
+	ProductVersion string `json:"operator"`
+	Matrix         matrix `json:"matrix"`
+}, installedVersion string) (*goversion.Version, error) {
```

Comment:
let's extract this struct to separate type.

Reply:
Done